vllm/v1/sample/sampler.py

```diff
@@ -243,7 +263,15 @@ def apply_logits_bias(
                             f"token_id {token_id} in logit_bias contains "
                             f"out-of-vocab token id. Vocabulary size: "
                             f"{vocab_size}")
-                    logits[i, token_id] += bias
+                        rows += [i]
+                        cols += [token_id]
+                        vals += [bias]
+
+        rows = list_to_gpu(rows, self.pin_memory, logits.device)
+        cols = list_to_gpu(cols, self.pin_memory, logits.device)
+        values = list_to_gpu(vals, self.pin_memory, logits.device)
```

Comment:
isn't bias on device already?

Reply:
it is a python float, https://github.com/vllm-project/vllm/blob/7594173c9e05aedb82e92ded0c1b04b2a7c684f2/vllm/v1/sample/metadata.py#L36